### ISSUE TITLE
CLI: panic: runtime error: index out of range [-1]

### ISSUE THREAD

**Issue by Dominik-K** [pulumi/pulumi#4688]
After pressing "space" on selected "No" at the "Do you want to perform this update?" question:

```
Do you want to perform this update?    [Use arrows to move, enter to select, type to filter]
================================================================================
The Pulumi CLI encountered a fatal error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v2.2.1
Go Version:       go1.14.2
Go Compiler:      gc
Architecture:     amd64
Operating System: darwin
Panic:            runtime error: index out of range [-1]

goroutine 1 [running]:
runtime/debug.Stack(0xc001842278, 0x5279e40, 0xc00153e000)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/debug/stack.go:24 +0x9d
main.panicHandler()
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/main.go:29 +0x6e
panic(0x5279e40, 0xc00153e000)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
gopkg.in/AlecAivazis/survey%2ev1.(*Select).OnChange(0xc001920820, 0x0, 0x0, 0x0, 0x0, 0x10, 0x6328f08, 0x0, 0x0, 0x0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.8/select.go:154 +0x6ad
gopkg.in/AlecAivazis/survey%2ev1.(*Select).Prompt(0xc001920820, 0x0, 0x0, 0x0, 0x0)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.8/select.go:234 +0x3b4
gopkg.in/AlecAivazis/survey%2ev1.Ask(0xc001842968, 0x1, 0x1, 0x5035860, 0xc001a84080, 0x0, 0x0, 0x0, 0xc000010018, 0xc00170e990)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.8/survey.go:148 +0x187
gopkg.in/AlecAivazis/survey%2ev1.AskOne(0x5709e60, 0xc001920820, 0x5035860, 0xc001a84080, 0x0, 0x0, 0x0, 0x0, 0x42, 0x0)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.8/survey.go:94 +0xeb
github.com/pulumi/pulumi/pkg/v2/backend.confirmBeforeUpdating(0x535b371, 0x6, 0x572c5c0, 0xc000fae600, 0xc00142e000, 0x4d, 0x80, 0x6328f08, 0x0, 0x0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/apply.go:168 +0x43c
github.com/pulumi/pulumi/pkg/v2/backend.PreviewThenPrompt(0x5712f60, 0xc0000480b8, 0x535b371, 0x6, 0x572c5c0, 0xc000fae600, 0xc000fae680, 0xc001028960, 0x54, 0xc00102e2a0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/apply.go:127 +0x24d
github.com/pulumi/pulumi/pkg/v2/backend.PreviewThenPromptThenExecute(0x5712f60, 0xc0000480b8, 0x535b371, 0x6, 0x572c5c0, 0xc000fae600, 0xc000fae680, 0xc001028960, 0x54, 0xc00102e2a0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/apply.go:199 +0x1ac
github.com/pulumi/pulumi/pkg/v2/backend/httpstate.(*cloudBackend).Update(0xc000fcb050, 0x5712f60, 0xc0000480b8, 0x572c5c0, 0xc000fae600, 0xc000fae680, 0xc001028960, 0x54, 0xc00102e2a0, 0x6328f08, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/httpstate/backend.go:812 +0xe6
github.com/pulumi/pulumi/pkg/v2/backend.UpdateStack(0x5712f60, 0xc0000480b8, 0x572c5c0, 0xc000fae600, 0xc000fae680, 0xc001028960, 0x54, 0xc00102e2a0, 0x6328f08, 0x0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/stack.go:82 +0xc4
github.com/pulumi/pulumi/pkg/v2/backend/httpstate.(*cloudStack).Update(0xc000fae600, 0x5712f60, 0xc0000480b8, 0xc000fae680, 0xc001028960, 0x54, 0xc00102e2a0, 0x6328f08, 0x0, 0x0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/backend/httpstate/stack.go:148 +0x96
main.newUpCmd.func1(0x6328f08, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7fffffff, 0x0, 0x6328f08, 0x0, ...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/up.go:133 +0xb0d
main.newUpCmd.func3(0xc0002adb80, 0xc00059bf10, 0x0, 0x1, 0x0, 0x0)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/up.go:372 +0x4e1
github.com/pulumi/pulumi/sdk/v2/go/common/util/cmdutil.RunResultFunc.func1(0xc0002adb80, 0xc00059bf10, 0x0, 0x1)
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/sdk/go/common/util/cmdutil/exit.go:112 +0x6b
github.com/spf13/cobra.(*Command).execute(0xc0002adb80, 0xc00059bf00, 0x1, 0x1, 0xc0002adb80, 0xc00059bf00)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002acdc0, 0x15, 0x0, 0x0)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /private/tmp/pulumi-20200514-76481-19mqsc6/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /private/tmp/pulumi-20200514-76481-19mqsc6/src/github.com/pulumi/pulumi/pkg/cmd/pulumi/main.go:48 +0x4d
```

**Comment by leezen** [comment]
Possibly related: https://github.com/pulumi/pulumi/issues/4677

**Comment by stack72** [comment]
Hi all

We believe this is closed via #4808

Sorry this wasn't closed out before

Paul
